spec/models/shelter_spec.rb

```diff
@@ -67,5 +67,67 @@
         expect(@shelter_1.pet_count).to eq(3)
       end
     end
+
+    describe "self.reverse_alphabetical" do
+      it "returns shelters in reverse alphabetical order" do
+        result = Shelter.order_by_reverse_alphabetical
+        expect(result).to eq([@shelter_2, @shelter_3, @shelter_1])
+      end
+    end
+
+    describe "#pending_apps" do
+      it "returns shelters with pets having pending applications" do
+        shelter_1 = Shelter.create(name: "Boulder Vally", city: "Boulder", foster_program: false, rank: 9)
+        shelter_2 = Shelter.create(name: "Run Away Haven", city: "New York", foster_program: false, rank: 5)
+
+        pet_1 = shelter_1.pets.create(name: "Bill", breed: "Boxer", age: 5, adoptable: false)
+        pet_2 = shelter_2.pets.create(name: "Marko", breed: "Hound", age: 3, adoptable: true)
+        pet_3 = shelter_2.pets.create(name: "Tiny", breed: "Pit", age: 2, adoptable: true)
+
+        johnny = Application.create!(name: 'Johnny', street_address: '111 rainbow rd', city: 'Iceblock', state: 'CO', zip_code: '80020',  reason_for_adoption: "I want dog", status: "Pending" )
+        kim = Application.create!(name: "Kim", street_address: "123 riverdale rd", city: "Thornton", state: "CO", zip_code: "80241", reason_for_adoption: "I want hound", status: "Approved")
+        susan = Application.create!(name: "Susan", street_address: "333 nighthawk ln", city: "Erie", state: "CO", zip_code: "45673", reason_for_adoption: "I am alone", status: "Rejected")
+
+        PetApplication.create!(pet: pet_1, application: johnny)
+        PetApplication.create!(pet: pet_2, application: kim)
+        PetApplication.create!(pet: pet_3, application: susan)
+
+        result = Shelter.pending_apps
+
+        expect(result).to include(shelter_1)
+        expect(result).not_to include(shelter_2)
+      end
+    end
+
+    it "orders shelters by name" do
+
+      ordered_shelters = Shelter.alphabetical_order.to_a
+
+      expected_order = [@shelter_1, @shelter_3, @shelter_2]
+
+      expect(ordered_shelters).to eq(expected_order)
```

Comment:
While it seems simple, this is a nice and robust test for this functionality! For model specs, you just want to make sure that at the lowest data level, you're testing the happy path (this test) and any additional edge cases that might be relevant. Sometimes, the logic is complex enough that additional tests are appropriate. This is a simple enough query that we don't need to handle alternative or sad paths (although you could if you tested that nothing breaks if you call this order method when there are no shelters... but that's overkill IMO)

The pending apps method has a bit more complexity to it and so you had to create more test data. You're testing the happy path, but I think you could make the test more robust by verifying that the method only returns ONE shelter back: `expect(shelters_with_pending_apps.length).to eq(1)` --> you could add this right after your assertion to make sure that not only does this return include that one shelter, but it's the only shelter that's given back. Your method is not erroneously returning non-pending apps.